### ISSUE TITLE
Proposed icon reorganization

### ISSUE THREAD

**Issue by Intigracy** [d3athrow/vgstation13#5465]
So I have an idea to make making inhands less of a pain in the ass. I just want to see if there's interest in me doing this before I take the time.

I plan to take all icons for objects (meaning things that can be picked up by people) and sorting them out much the same way that they are for the base icon now, except that it'll be slightly different.

There will be 3 icon states per item, 4 for clothing. Item icons (on the ground), 2 inhand icons, and on mob icons (clothing only). These will all be in the <b>same file</b>. 
I'd still be setting up the dmis so that headgear is in one dmi, uniforms in another, guns in their own, etc. Although it will probably get split even further (such as by department for clothes) to lower the size of the already giant uniform.dmi, since it's likely that it will (it definitely will) hit the icon_state per dmi limit if it quadruples in size.

Icons in the dmis will be named, in the case of the cakehat for example, "cakehat" for the icon, "cakehat_l" and "cakehat_r" for the inhands, and "cakehat_om" for onmob. Obviously non-clothing items would lack an _om icon.

The way the code aspect of it would work is this:
The icon, icon_state, and item_state variables will remain. Icon will deal with the dmi, as it does now, icon_state with the object icon itself, and item_state with the inhand, as I'm pretty sure it does now. Clothing layers will automatically apply based on the _om icon_state within the same dmi as is specified in var/Icon. 
   * This means we retain the ability to assign a seperate item_state for the bunch of items that share a common one (like jumpsuits using the default gray) without having to copy it a bunch of times for each and every item using it.
   * Due to the way this is set up, admins will actually be able to edit clothing/weapon icons without having to play with overlays (which doesn't even work anymore). All they'll need is a dmi with an icon, inhand, and onmob icon inside to edit the icon variables to.

This is probably confusing to read, so here's a mockup.

![](http://puu.sh/j8ug4/76098ae291.png)

![](http://puu.sh/j8uxz/2d51d0a081.png)

TL;DR, lots of little dmi's with all same item icons condensed rather than a few big dmi's with everything seperated by type (inhands are currently ONLY with other inhands, which are seperate from the item's icon_state, which are both seperate from the actual onmob sprite in the case of clothes).

**Comment by Intigracy** [comment]
This would fix [4331].

**Comment by ComicIronic** [comment]
It would increase redundancy on items that share in-hand icons.

**Comment by Intigracy** [comment]
Are there any items that share inhand icons that it wouldn't make sense to have them in the same dmi?

**Comment by ComicIronic** [comment]
A massive number of items use the metal sheet in-hand.

**Comment by Intigracy** [comment]
Maybe someday I will get around to this
